packages/suite/src/reducers/wallet/graphReducer.ts

```diff
@@ -104,9 +105,12 @@ const graphReducer = (state: State = initialState, action: WalletAction | SuiteA
             case GRAPH.SET_SELECTED_VIEW:
                 draft.selectedView = action.payload;
                 break;
-            case ACCOUNT.REMOVE:
-                remove(draft, action.payload);
+            case accountsActions.removeAccount.type: {
+                if (accountsActions.removeAccount.match(action)) {
```

Comment:
For runtime yes. For TS no. 

Consider this as type guard just for TS, this is best way how to let TS know what action it is. All other ways like type casting etc. are problematic (it's quite well described in RTK docs) and this is super easy and always works.

In all other places like middlewares we just removed case from switch and place if directly before switch itself, but here in legacy reducer I really don't want to break pattern where you have only one switch with cases.

Reply:
Yeah, I understood that it type guards, it's just there is only 1 line of code inside that `if`, and nothing is red without it, hence me asking if it might be a bit extra 😅 Still, it's definitely nothing bad!

Reply:
Actually, the payload is not typed properly if you don't use the matching function for action types. It is probably just needed until we get rid of these traditional (switch/case) reducers, because toolkit takes care of this already, but right now, the payload is typed as never without this match function. But if we match it, payload types are correct without TS screaming. 😱